pkg/tasks/prometheusadapter.go

```diff
@@ -159,6 +150,38 @@ func (t *PrometheusAdapterTask) Run(ctx context.Context) error {
 		}
 	}
 	{
+		cmD, err := t.factory.PrometheusAdapterConfigMapDedicated()
```

Comment:
Also, random idea, what if both ConfigMaps had the same name? This way we would not have to go through the update/delete.
Note that in this approach we would have to add an annotation to the deployments with a hash of the config to trigger redeployments.

Reply:
Not sure I understand the advantage of this approach :thinking: 

Wouldn't we still have to update the ConfigMap and pass data to `t.factory.PrometheusAdapterDeployment`?
Part of the motivation to change the ConfigMap name in the prometheus-adapter Deployment was that this makes it immediately obvious which config is used.

Reply:
If I'm thinking correctly the advantage would be that instead of a Create&Delete we would only need to call CreateOrUpdate, which would make this part of the code consistent with other areas of the operator.

However, if one of the goals of the change was to make it obvious what configuration was being used then this would be the way to go.

Reply:
I agree that having 2 different configmaps is more obvious.
Semi-related question: why not have the 2 configmaps always present and only deal with selecting the correct one in the adapter deployment? The advantage would be less code branches.